### ISSUE TITLE
Retry triggered on success?

### ISSUE THREAD

**Issue by hffmnn** [arikis/Overdrive#4]
**Overdrive:** `0.2.1`
**Package manager (SPM, Carthage, CocoaPods, Manual):** `Cocoapods 1.1`
**Xcode:** `8.2 beta 2`
**Platform:** `iOS 10`

Running a simple, non-failable task with `retry`, retries even if the task finishes without error.
```
let inlineTask = InlineTask { (_) in
    print("inlineTask")
}.retry(2)

queue.add(task: inlineTask)
```

prints `inlineTask` 3 Times to the console, even if the documentation of `retry` states:
> If the task finishes with error, it will be executed again until retry count becomes zero.

**Comment by ManWithBear** [comment]
Please, checkout my fix in #5

**Comment by arikis** [comment]
Thanks for tracking this one down. Finish logic still needs some work and this is a step in the right direction.
